### PR TITLE
hwmv2: Added Board Support for Starfive VisionFive2 RISC-V Single Board Computer

### PR DESCRIPTION
**Pull request by pfarwsi**
These list of patches contain Board support files for running Zephyr on the Starfive VisionFive2 RISC-V Single Board Computer based on JH7110 SOC. It's a basic support, no drivers other than UART have been tested.

@ycsin @aescolar

### CODE REVIEW

File: boards/starfive/visionfive2_jh7110/doc/img/Visionfive2.jpg
Comment (nordicjm) [inline boards/starfive/visionfive2_jh7110/doc/img/Visionfive2.jpg:1]: Make the board image smaller (e.g. webp)

File: boards/starfive/visionfive2_jh7110/doc/index.rst
Comment (nordicjm) [inline boards/starfive/visionfive2_jh7110/doc/index.rst:26]: the soc is jh7110? So the board should be `visionfive2` then if there is only a single soc, that is the name that is used to build, if there is multiple socs then it would be built using e.g. `visionfive2/jh7110`

File: boards/starfive/visionfive2_jh7110/visionfive2_jh7110.yaml
Comment (nordicjm) [inline boards/starfive/visionfive2_jh7110/visionfive2_jh7110.yaml:1]: as above

File: boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig
Comment (nordicjm) [inline boards/starfive/visionfive2_jh7110/visionfive2_jh7110_defconfig:3]:
```suggestion
```

File: soc/starfive/jh71xx/Kconfig.soc
Comment (nordicjm) [inline soc/starfive/jh71xx/Kconfig.soc:28]:
```suggestion
config SOC
	default "jh7100" if SOC_JH7100
	default "jh7110" if SOC_JH7110
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/starfive/visionfive2_jh7110/doc/index.rst
Comment (pfarwsi) [reply boards/starfive/visionfive2_jh7110/doc/index.rst]: Modified as per this suggestion. The board name is now "visionfive2", updated the documentation according to that

File: boards/starfive/visionfive2_jh7110/Kconfig.defconfig
Comment (pfarwsi) [reply boards/starfive/visionfive2_jh7110/Kconfig.defconfig]: What would be the correct way to do that?

Comment (kanakshilledar) [reply boards/starfive/visionfive2_jh7110/Kconfig.defconfig]: We have incorporated the changes.

**Review by ycsin — CHANGES_REQUESTED** [review]
commit 3:

- title and body should just mention the introduction of a new board
- CODEOWNERS and MAINTAINERS.yml changes should be in another commit (probably the last commit)

commit 4:
- should be squashed with the commit that introduced the 7110 soc

commit 5:
- should be squashed with the commit that introduced that yaml file

___

We generally dont make another commit to just "address maintainers' comments". Rather these requested changes should be made and squashed with the commit that introduced the changes initially where it makes sense.

**Comment by kanakshilledar** [comment]
Updated the commits according to the errors thrown by compliance checks.

**Comment by kanakshilledar** [comment]
@ycsin I have updated the commits according to your suggested changes.

File: boards/starfive/visionfive2/board.cmake
Comment (nordicjm) [inline boards/starfive/visionfive2/board.cmake:1]: delete this empty file

File: CODEOWNERS
Comment (nordicjm) [inline CODEOWNERS:122]: This folder doesn't exist

**Review by nordicjm — DISMISSED** [review]

**Comment by kanakshilledar** [comment]
@nordicjm I have incorporated those suggested changes.

File: boards/starfive/visionfive2/Kconfig.defconfig
Comment (ycsin) [inline boards/starfive/visionfive2/Kconfig.defconfig:1]: 2024?

File: MAINTAINERS.yml
Comment (ycsin) [inline MAINTAINERS.yml:2869]: should we introduce a new entry for `Starfive Platforms`?

**Review by ycsin — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (kanakshilledar) [reply MAINTAINERS.yml:2869]: We are planning to add it under the RISCV Arch area.

Comment (pfarwsi) [reply MAINTAINERS.yml:2869]: @ycsin Do you think we should create a new entry for Starfive or keep it as it is?

**Review by kanakshilledar — APPROVED** [review]

**Comment by kanakshilledar** [comment]
@ycsin We feel that once there are multiple boards from StarFive we can create a separate area for the same, till then let it be in the RISCV Arch area.

**Review by ycsin — DISMISSED** [review]
I think the dts, soc and board changes looks fine, but not sure if it makes sense to add new boards under RISCV arch, will leave that to the maintainers, cc @fkokosinski 

It might make sense to move out those sifive and stuffs into `SiFive Platforms` or `RISCV Platforms`, leaving just the qemu targets under `RISCV arch`

**Review by cfriedt — DISMISSED** [review]
LGTM

**Comment by kanakshilledar** [comment]
I would like to know the next process after these approvals. Excited to get my PR merged in `main` 😄

File: boards/starfive/visionfive2/doc/index.rst
Comment (kartben) [inline boards/starfive/visionfive2/doc/index.rst:32]: This is redundant with previous paragraph (and it should use zephyr-app-commands anyway)

File: boards/starfive/visionfive2/doc/index.rst
Comment (kartben) [inline boards/starfive/visionfive2/doc/index.rst:41]: It would be good to explain *why* one needs to do this

File: boards/starfive/visionfive2/doc/index.rst
Comment (kartben) [inline boards/starfive/visionfive2/doc/index.rst:39]: This seems to imply one would download the tool at the root of their Zephyr source tree?

File: boards/starfive/visionfive2/doc/index.rst
Comment (kartben) [inline boards/starfive/visionfive2/doc/index.rst:56]: Use double back quotes or the `:file:` role here and elsewhere when your referring to filenames. This makes them much easier to read

File: MAINTAINERS.yml
Comment (kartben) [inline MAINTAINERS.yml:2863]: Are you already planning to help with all RISC-V related work in the project? I.e. review the incoming PRs etc?
Given this looks like your first contribution, it might be a bit premature IMO
Not sure who's the other person but I'm assuming (perhaps wrongly) the same might be true

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/starfive/visionfive2/doc/index.rst
Comment (pfarwsi) [reply boards/starfive/visionfive2/doc/index.rst]: Explained

File: boards/starfive/visionfive2/doc/index.rst
Comment (pfarwsi) [reply boards/starfive/visionfive2/doc/index.rst:35]: Doesn't matter if it's in root of zephyr source tree or not. The tool can be cloned anywhere, all that it is needed for is to create spl header for zephyr binary.

File: MAINTAINERS.yml
Comment (pfarwsi) [reply MAINTAINERS.yml:2863]: Yes we have already planned for the RISC-V related work going forward which would be adding more features for this board, looking into issues and also review the incoming PR's. Since RISC-V area is quite new for everyone and needs more and more support, we thought of contributing this small piece of work to the community :). There would be more and more work coming forward in this area. This is a joint activity between @pfarwsi and @kanakshilledar.

File: dts/riscv/starfive/jh7110-visionfive-v2.dtsi
Comment (fkokosinski) [inline dts/riscv/starfive/jh7110-visionfive-v2.dtsi:16]: `StarFive VisionFive V2` is the name of the board, right? Not the SoC.

File: dts/riscv/starfive/jh7110-visionfive-v2.dtsi
Comment (fkokosinski) [inline dts/riscv/starfive/jh7110-visionfive-v2.dtsi:195]: `uart0` and `uart1` should be selected in the board dts, unless we're 100% sure that they need to be used for this SoC.

File: MAINTAINERS.yml
Comment (fkokosinski) [reply MAINTAINERS.yml:2863]: I have to agree with @kartben on this one. I've noticed that the usual path people take to be responsible for an area of Zephyr is as follows:
1. contribute & in general - interact with the community (you're here)
2. start reviewing other people's code
3. become a collaborator
4. become a maintainer

We're of course more than happy to have you and your contributions here. That being said, being responsible for the RISC-V architecture as a whole means you need to have the broader picture, be aware of all the different conventions the project conforms to and so on. And I'm just not sure if that's true for you, having only one PR under your belt.

IMO a smaller area would be better for a new member of the community, i.e. a StarFive-specific one (e.g. see the ITE platforms area) which @ycsin proposed here: https://github.com/zephyrproject-rtos/zephyr/pull/69814/files#r1514001131.

I'm unresolving this thread for now, as this IMO needs more discussion.

**Review by fkokosinski — CHANGES_REQUESTED** [review]

File: dts/riscv/starfive/jh7110-visionfive-v2.dtsi
Comment (pfarwsi) [reply dts/riscv/starfive/jh7110-visionfive-v2.dtsi]: uart0 and uart1 disabled from dtsi and enabled from dts.

File: MAINTAINERS.yml
Comment (cfriedt) [inline MAINTAINERS.yml:2863]: It will take a bit of time before you can become arch maintainers as pointed out by @fkokosinski 

Review roles here 

https://docs.zephyrproject.org/latest/project/project_roles.html

```suggestion
```

**Review by cfriedt — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (pfarwsi) [reply MAINTAINERS.yml:2863]: Ok i have incorporated the changes @fkokosinski @cfriedt

**Review by cfriedt — DISMISSED** [review]

**Review by ycsin — COMMENTED** [review]
last commit title and body should be updated

**Comment by kanakshilledar** [comment]
@ycsin I have incorporated the suggested changes in the last commit message title and body

**Review by fkokosinski — CHANGES_REQUESTED** [review]
Last nit, LGTM otherwise.

**Review by nordicjm — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by ycsin — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**
